jquery.jsticky.js

```diff
@@ -43,56 +43,75 @@
 
-      var $this          = $(this),
-          thisHeight     = $this.outerHeight(),
-          thisWidth      = $this.outerWidth(),
-          topSpacing     = settings.topSpacing,
-          zIndex         = settings.zIndex,
-          pushPoint      = $this.offset().top - topSpacing, // Point at which the sticky element starts pushing
-          placeholder    = $('<div></div>').width(thisWidth).height(thisHeight).addClass('sticky-placeholder'), // Cache a clone sticky element
-          stopper        = settings.stopper,
-          $window        = $(window);
+      var $this = $(this);
+      var thisHeight = $this.outerHeight();
+      var thisWidth = $this.outerWidth();
+      var topSpacing = settings.topSpacing;
+      var zIndex = settings.zIndex;
+      var pushPoint = $this.offset().top - topSpacing; // Point at which the sticky element starts pushing
+      var placeholder = $('<div></div>').width(thisWidth).height(thisHeight).addClass('sticky-placeholder'); // Cache a clone sticky element
+      var stopper = settings.stopper;
+      var $window = $(window);
 
       function stickyScroll() {
 
         var windowTop  = $window.scrollTop(); // Check window's scroll position
-        
+        var stopPoint = stopper;
+        var parentWidth = $this.parent().width();
+
+        placeholder.width(parentWidth)
+
         if ( hasStopper && typeof stopper === 'string' ) {
-          var stopperTop = $(stopper).offset().top,
-              stopPoint  = (stopperTop - thisHeight) - topSpacing;
-        } else if (hasStopper && typeof stopper === 'number') {
-          var stopPoint = stopper;
+          var stopperTop = $(stopper).offset().top;
+          stopPoint  = (stopperTop - thisHeight) - topSpacing;
         }
 
         if (pushPoint < windowTop) {
           // Create a placeholder for sticky element to occupy vertical real estate
+          if(settings.stickyClass)
+            $this.addClass(settings.stickyClass);
+
           $this.after(placeholder).css({
             position: 'fixed',
-            top: topSpacing
+            top: topSpacing,
+            width: parentWidth
           });
-          
+
           if (hasIndex) {
-            $this.css({ zIndex: zIndex });
+            $this.css({
+              zIndex: zIndex
+            });
           }
 
           if (hasStopper) {
             if (stopPoint < windowTop) {
               var diff = (stopPoint - windowTop) + topSpacing;
-              $this.css({ top: diff });
+              $this.css({
+                top: diff
+              });
             }
           }
         } else {
+          if(settings.stickyClass)
+            $this.removeClass(settings.stickyClass);
+
           $this.css({
             position: 'static',
             top: null,
-            left: null
+            left: null,
+            width: 'auto'
           });
 
           placeholder.remove();
         }
-      };
+      }
+
+      if($window.innerHeight() > thisHeight) {
```

Comment:
I think we create function that checks the window height is bound to load and resize. The function can determine if the sticky plugin should be bound or unbound as a result of the window height.

The way it's written, if the window size starts out too small, the plugin won't work unless the user refreshes.